WordPress/Classes/Utility/CoreDataHelper.swift

```
@@ -0,0 +1,149 @@
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func allObjects(matchingPredicate predicate: NSPredicate? = nil, sortedBy descriptors: [NSSortDescriptor]? = nil)  -> [T] {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.sortDescriptors = descriptors
+
+        return loadObjects(withFetchRequest: request)
+    }
+
+    /// Returns the number of entities found that match with a given predicate.
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func countObjects(matchingPredicate predicate: NSPredicate? = nil) -> Int {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.includesSubentities = false
+        request.predicate = predicate
+
+        var result = 0
+
+        do {
+            result = try context.countForFetchRequest(request)
+        } catch {
+            DDLogSwift.logError("Error counting objects [\(T.entityName)]")
+        }
+
+        return result
+    }
+
+    /// Deletes the specified Object Instance
+    ///
+    func deleteObject(object: T) {
+        do {
+            context.deleteObject(object)
+            try context.save()
+        } catch {
+            DDLogSwift.logError("Error deleting entity [\(T.entityName)]")
+        }
+    }
+
+    /// Deletes all of the NSMO instances associated to the current kind
+    ///
+    func deleteAllObjects() {
+        let request = newFetchRequest()
+        request.includesPropertyValues = false
+        request.includesSubentities = false
+
+        do {
+            let objects = loadObjects(withFetchRequest: request)
+            for object in objects  {
+                context.deleteObject(object)
+            }
+
+            try context.save()
+        } catch {
+            DDLogSwift.logError("Error deleting all entities of kind [\(T.entityName)]")
```

Silent error failure.